src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -1309,6 +1310,309 @@ bool ParserCodec::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
     return true;
 }
 
+const char * ParserCollation::valid_collations[] = {
```

Comment:
It's a bit overkill, need just expect `isWordCharASCII`

Reply:
> It's a bit overkill, need just expect `isWordCharASCII`

I feel confused. If I try to parse `... COLLATE NOT NULL`, will  `isWordCharASCII` regard the `NOT` as `false` ?

Reply:
Good point! In that case maybe just check suffix https://github.com/ClickHouse/ClickHouse/pull/36271/commits/f6f58f7609600f2ccc8479e015ac00a794894846

Reply:
> Good point! In that case maybe just check suffix [f6f58f7](https://github.com/ClickHouse/ClickHouse/commit/f6f58f7609600f2ccc8479e015ac00a794894846)

Good Job